akka-http/src/main/scala/akka/http/javadsl/server/directives/HeaderDirectives.scala

```diff
@@ -34,7 +35,11 @@ abstract class HeaderDirectives extends FutureDirectives {
    * @group header
    */
   def checkSameOrigin(allowed: HttpOriginRange, inner: jf.Supplier[Route]): Route = RouteAdapter {
```

Comment:
In general it is about creating an interface in the javadsl.model package, in the equivalent place with the same name, and make the scala one extend the java one. MiMa (our binary compatibility checker) may get triggered by this, but I believe this is safe here since no one is meant to be extending the Default class outside from Akka's model itself.